build_tools/bazel/iree_check_test.bzl

```diff
@@ -67,7 +67,7 @@ def iree_check_test(
     native_test(
         name = name,
         args = [
-            "--driver=%s" % driver,
+            "--device=%s" % driver,
```

Comment:
We should also update the arg name from `driver` to `device`, including comment updates 
```
      driver: driver to run the module with. This can be omitted to test only
          compilation, but consider omiting the driver as a hacky abuse of the
          rule since compilation on its own not use iree-check-module.
```

looks like a few files under `build_tools/` need similar updates

Reply:
I think this use of driver is fine as it's specifying the driver and not a device - it maps to the driver= test filter flag and which driver module to link in, which we then use to derive the --device flag. There's still probably a few hiding in here but I tried not to touch anything that ended up in driver-related stuff as going driver->device is safe but device->driver isn't (don't want people to set a benchmark suite rule for a particular device path and then have to reverse engineer the driver from that with cmake string manipulation).

Reply:
Still a little confusing IMO, but fine to proceed and continue cleaning up later.